src/otx/algo/detection/rtdetr.py

```diff
@@ -0,0 +1,438 @@
+        if self.image_size is None:
+            raise ValueError(self.image_size)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=self.image_size,
+            mean=self.mean,
+            std=self.std,
+            resize_mode="standard",
+            swap_rgb=False,
+            via_onnx=False,
+            onnx_export_configuration={
+                "input_names": ["images"],
+                "output_names": ["bboxes", "labels", "scores"],
+                "dynamic_axes": {
+                    "images": {0: "batch"},
+                    "boxes": {0: "batch", 1: "num_dets"},
+                    "labels": {0: "batch", 1: "num_dets"},
+                    "scores": {0: "batch", 1: "num_dets"},
+                },
+                "autograd_inlining": False,
+                "opset_version": 16,
+            },
+            output_names=["bboxes", "labels", "scores"],
+        )
+
+    @property
+    def _optimization_config(self) -> dict[str, Any]:
+        """PTQ config for RT-DETR."""
+        return {"model_type": "transformer"}
+
+
+class OTXRtDetr18(OTXRtDetr):
+    load_from = (
+        "https://github.com/lyuwenyu/storage/releases/download/v0.1/rtdetr_r18vd_5x_coco_objects365_from_paddle.pth"
+    )
+
+    def _build_model(self, num_classes: int) -> nn.Module:
+        backbone = PResNet(
+            depth=18,
+            pretrained=True,
+            freeze_at=-1,
+            return_idx=[1, 2, 3],
+            num_stages=4,
+        )
+        encoder = HybridEncoder(
+            in_channels=[128, 256, 512],
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            expansion=0.5,
+            dim_feedforward=1024,
+            eval_spatial_size=self.image_size[2:],
+        )
+        decoder = RTDETRTransformer(
+            num_classes=num_classes,
+            num_decoder_layers=3,
+            feat_channels=[256, 256, 256],
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            eval_spatial_size=self.image_size[2:],
+        )
+        criterion = RTDetrCriterion(
+            weight_dict={"loss_vfl": 1.0, "loss_bbox": 5, "loss_giou": 2},
+            num_classes=num_classes,
+            gamma=2.0,
+            alpha=0.75,
+        )
+        optimizer_configuration = [
+            {"params": "^(?=.*backbone)(?=.*norm).*$", "weight_decay": 0.0, "lr": 0.00001},
+            {"params": "^(?=.*backbone)(?!.*norm).*$", "lr": 0.00001},
+            {"params": "^(?=.*(?:encoder|decoder))(?=.*(?:norm|bias)).*$", "weight_decay": 0.0},
+        ]
+
+        return RTDETR(
+            backbone=backbone,
+            encoder=encoder,
+            decoder=decoder,
+            num_classes=num_classes,
+            criterion=criterion,
+            optimizer_configuration=optimizer_configuration,
+        )
+
+
+class OTXRtDetr50(OTXRtDetr):
```

Comment:
I'd like to suggest renaming it to `RTDETRRes50` referring to the official repo.
https://github.com/lyuwenyu/RT-DETR?tab=readme-ov-file#-implementations

Reply:
I end up with RTDETR50
https://github.com/openvinotoolkit/training_extensions/pull/3741/commits/2d5cef1ef2bbc576bebd10fd1a96e812d6ef0548